documentation/operators/src/nodes/maintenance.md

```diff
@@ -0,0 +1,699 @@
+
+Note: Cosmovisor will swap the `nyxd` binary, but you'll need to already have the `libwasmvm.x86_64.so` in place.
+
+#### Common reasons for your validator being jailed
+
+The most common reason for your validator being jailed is that your validator is out of memory because of bloated syslogs.
+
+Running the command `df -H` will return the size of the various partitions of your VPS.
+
+If the `/dev/sda` partition is almost full, try pruning some of the `.gz` syslog archives and restart your validator process.
+
+## Moving a node
+
+In case of a need to move a node from one machine to another and avoiding to lose the delegation, here are few steps how to do it.
+
+The following examples transfers a mixnode (in case of other nodes, change the `mixnodes` in the command for the `<NODE>` of your desire.
+
+* Pause your node process.
+
+Assuming both machines are remote VPS.
+
+* Make sure your `~/.ssh/<YOUR_KEY>.pub` is in both of the machines `~/.ssh/authorized_keys` file
+* Create a `mixnodes` folder in the target VPS. Ssh in from your terminal and run:
+```
+# in case none of the nym configs was created previously
+mkdir ~/.nym
+
+#in case no nym mixnode was initialized previously
+mkdir ~/.nym/mixnodes
+```
+* Move the node data (keys) and config file to the new machine by opening a local terminal (as that one's ssh key is authorized in both of the machines) and running:
+```
+scp -r -3 <SOURCE_USER_NAME>@<SOURCE_HOST_ADDRESS>:~/.nym/mixnodes/<YOUR_ID> <TARGET_USER_NAME>@<TARGET_HOST_ADDRESS>:~/.nym/mixnodes/
+```
+* Re-run init (remember that init doesn't overwrite existing keys) to generate a config with the new listening address etc.
+* Change the node smart contract info via the wallet interface. Otherwise the keys will point to the old IP address in the smart contract, and the node will not be able to be connected, and it will fail up-time checks.
+* Re-run the node from the new location. 
+
+## VPS Setup and Automation
+### Configure your firewall
+Although your `<NODE>` is now ready to receive traffic, your server may not be. The following commands will allow you to set up a firewall using `ufw`.
+
+```
+# check if you have ufw installed
+ufw version
+
+# if it is not installed, install with
+sudo apt install ufw -y
+
+# enable ufw
+sudo ufw enable
+
+# check the status of the firewall
+sudo ufw status
+```
+
+Finally open your `<NODE>` p2p port, as well as ports for ssh and ports for verloc and measurement pings:
+
+```
+# for mixnode
+sudo ufw allow 1789,1790,8000,22/tcp
+
+# for gateway
+sudo ufw allow 1789,22,9000/tcp
+
+# for network requester
+sudo ufw allow 22,9000/tcp
+
+# for validator
+sudo ufw allow 1317,26656,26660,22,80,443/tcp
+```
+
+Check the status of the firewall:
+```
+sudo ufw status
+```
+
+For more information about your node's port configuration, check the [port reference table](https://nymtech.net/docs/nodes/gateway-setup.html#ports) below.
```

Comment:
Why specifically refer to the gateway ports? Maybe we can make a generic ports table